spanner/read.go

```diff
@@ -138,10 +142,32 @@ var _ rowIterator = (*RowIterator)(nil)
 // there are no more results. Once Next returns Done, all subsequent calls
 // will return Done.
 func (r *RowIterator) Next() (*Row, error) {
+	mt := r.meterTracerFactory.createBuiltinMetricsTracer(r.ctx)
 	if r.err != nil {
 		return nil, r.err
 	}
-	for len(r.rows) == 0 && r.streamd.next() {
+	// Start new attempt if we don't have one
+	if mt.currOp.currAttempt == nil {
+		mt.currOp.incrementAttemptCount()
+		mt.currOp.currAttempt = &attemptTracer{
+			startTime: time.Now(),
+		}
+	}
+	defer func() {
+		if mt.method != "" {
+			// record the attempt completion
+			if mt.currOp.currAttempt != nil {
```

Comment:
When is `mt.currOp.currAttempt == nil`? On line 152 above, mt.currOp.currAttempt is set to something that is not nil. Apparently, something else can set it back to nil before we get here. What is that?